npc/ipset/ipset.go

```diff
@@ -67,3 +78,38 @@ func doExec(args ...string) error {
 	}
 	return nil
 }
+
+// Reference-counting
+type key struct {
+	ipsetName Name
+	entry     string
+}
+
+// note no locking is required as all operations are serialised in the controller
```

Comment:
Not sure if this matters or not -- still looking at the code and the Go memory model -- but:
```
func (npc *controller) onNewNsSelector(selector *selector) error
``` 
does not lock like other method calls which do:
```
	npc.Lock()
	defer npc.Unlock()
```